tensorboard/webapp/metrics/views/card_renderer/utils.ts

```diff
@@ -29,3 +30,52 @@ export function getDisplayNameForRun(
 
   return displayName;
 }
+
+/**
+ * Partitions runs into pseudo runs when its points have non-monotonically increasing
+ * steps.
+ */
+export function partitionSeries(series: PartialSeries[]): PartitionedSeries[] {
+  const partitionedSeries: PartitionedSeries[] = [];
+  for (const datum of series) {
+    const currentPartition: Array<Omit<
+      PartitionedSeries,
+      'partitionSize' | 'partitionIndex'
+    >> = [];
+    let lastXValue = datum.points[0]?.x ?? 0;
```

Comment:
I just tried writing a scalar `step=-1`, and it seems negative steps work in existing TensorBoard.  Should this `lastXValue` be `-Infinity`?

Reply:
Just wondering, how would that manifest as a bug?